controllers/metallb_controller.go

```diff
@@ -269,3 +241,133 @@ func valueWithDefault(name string, def int) (int, error) {
 	}
 	return def, nil
 }
+
+func patchPrometheusValues(chartValues map[string]interface{}, client client.Client) error {
+	metricsPort, err := valueWithDefault("METRICS_PORT", 7472)
+	if err != nil {
+		return err
+	}
+	// We shouldn't spam the api server trying to apply PodMonitors if the resource isn't installed.
+	var enablePodMonitor bool
+	if podMonitorAvailable(client) && os.Getenv("DEPLOY_PODMONITORS") == "true" {
+		enablePodMonitor = true
+	}
+	chartValues["prometheus"] = map[string]interface{}{
+		"metricsPort": metricsPort,
+		"podMonitor": map[string]interface{}{
+			"enabled": enablePodMonitor,
+		},
+	}
+	return nil
+}
+
+func patchControllerValues(chartValues map[string]interface{}, config *metallbv1beta1.MetalLB, isOpenShift bool) {
+	logLevel := metallbv1beta1.LogLevelInfo
+	if config.Spec.LogLevel != "" {
+		logLevel = config.Spec.LogLevel
+	}
+
+	ctrlImage := os.Getenv("CONTROLLER_IMAGE")
```

Comment:
Can this (and the other images) be not set?
I'd keep it simple and fail if the needed env variables are not set